### PR TITLE
Spring data jpa improvements

### PR DESCRIPTION
**Pull request by esidarapp**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -34,6 +44,36 @@
...
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-web</artifactId>
```
Comment (Elena-Bruyako) [inline pom.xml]: Why do you need this dependency?

File: pom.xml [hunk 2]
```diff
@@ -34,6 +44,36 @@
...
+			<version>6.1.6</version>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+			<version>3.3.0</version>
```
Comment (Elena-Bruyako) [inline pom.xml]:
```suggestion
```

File: src/main/java/com/example/spring/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/model/Book.java]: Don't use `@Data` for entities
```suggestion
@Getter
@Setter
```

File: src/main/java/com/example/spring/model/Book.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/model/Book.java:16]: You should implement soft-delete approach 
```suggestion
@Entity
@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id = ?")
@SQLRestriction(value = "is_deleted=false")
```

File: src/main/java/com/example/spring/service/BookService.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+public interface BookService {
+    BookDto save(CreateBookRequestDto requestDto);
+
+    List<BookDto> findAll();
+
+    BookDto findById(Long id);
+
+    BookDto updateById(Long id, CreateBookRequestDto requestDto);
+
+    BookDto deleteById(Long id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/service/BookService.java]:
```suggestion
    void deleteById(Long id);

```

File: src/main/java/com/example/spring/service/BookServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        return bookMapper.toDto(book);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book "
+                        + "by id " + id)
+        );
+        book.setPrice(requestDto.getPrice());
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/service/BookServiceImpl.java]: Look like the Mapper logic here
`bookMapper.updateBookFromDto(bookDto, book);`

File: src/main/java/com/example/spring/service/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        book.setAuthor(requestDto.getAuthor());
+        book.setIsbn(requestDto.getIsbn());
+        book.setTitle(requestDto.getTitle());
+        book.setDescription(requestDto.getDescription());
+        book.setCoverImage(requestDto.getCoverImage());
+        bookRepository.save(book);
+        return bookMapper.toDto(book);
+    }
+
+    @Override
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/service/BookServiceImpl.java:53]:
```suggestion
    @Override
    public void deleteById(Long id) {
        bookRepository.deleteById(id);
    }
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve it.

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**